x-pack/plugin/logsdb/src/test/java/org/elasticsearch/xpack/logsdb/LogsdbIndexModeSettingsProviderTests.java

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.logsdb;
+
+import org.elasticsearch.cluster.metadata.ComposableIndexTemplate;
+import org.elasticsearch.cluster.metadata.ComposableIndexTemplateMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.metadata.Template;
+import org.elasticsearch.common.compress.CompressedXContent;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Map;
+
+public class LogsdbIndexModeSettingsProviderTests extends ESTestCase {
+
+    public static final String DEFAULT_MAPPING = """
+        {
+            "_doc": {
+                "properties": {
+                    "@timestamp": {
+                        "type": "date"
+                    },
+                    "message": {
+                        "type": "keyword"
+                    },
+                    "host.name": {
+                        "type": "keyword"
+                    }
+                }
+            }
+        }
+        """;
+
+    public void testLogsDbDisabled() throws IOException {
+        final LogsdbIndexModeSettingsProvider provider = new LogsdbIndexModeSettingsProvider(
+            Settings.builder().put("cluster.logsdb.enabled", false).build()
+        );
+
+        final Settings additionalIndexSettings = provider.getAdditionalIndexSettings(
+            null,
+            "logs-apache-production",
+            false,
+            Metadata.EMPTY_METADATA,
+            Instant.now().truncatedTo(ChronoUnit.SECONDS),
+            Settings.EMPTY,
+            List.of(new CompressedXContent(DEFAULT_MAPPING))
+        );
+
+        assertIndexMode(additionalIndexSettings, null);
```

Comment:
In this case the returned settings should be empty. Maybe assert whether `additionalIndexSettings.isEmpty()` is true instead?


Maybe all `assertIndexMode(additionalIndexSettings, null);` imstances can be replaced with `assertTrue(additionalIndexSettings.isEmpty())`